tests/Framework/Feature/Identification/TenantIdentificationEventsTest.php

```diff
@@ -45,6 +45,31 @@ public function empty_tenant_is_forced()
         $this->assertEquals(0, $identified);
     }
 
+    /** @test */
+    public function old_tenant_in_switched()
+    {
+        $oldTenant = null;
+        $this->events->listen(Switched::class, function (Switched $switched) use (&$newTenant, &$oldTenant) {
+            $newTenant = $switched->tenant;
+            $oldTenant = $switched->oldTenant;
+        });
+
+        $tenant1 = $this->mockTenant();
+        $tenant2 = $this->mockTenant();
+
+        $this->assertNotEquals($tenant1, $tenant2);
+
+        $this->environment->setTenant($tenant1);
+
+        $this->assertNull($oldTenant);
+        $this->assertEquals($tenant1, $newTenant);
+
+        $this->environment->setTenant($tenant2);
+
+        $this->assertEquals($tenant1, $oldTenant);
+        $this->assertEquals($tenant2, $newTenant);
+    }
+
```

Comment:
This test also need to test for the scenario that no initial tenant is identified, so that it confirms `$oldTenant` is actually `null` in some situations.

Reply:
I don't really understand what you mean, there is already a `$this->assertNull($oldTenant);` on the first event

The test looks complete to me

Reply:
I missed that, you're right 👍